### DISCUSSION TITLE
New trigger showing group and index of sound currently playing. Expansion of possibilities for working with sound

### DISCUSSION THREAD

**Discussion by Kiwi1444** [ikemen-engine/Ikemen-GO#1902] [Feature Request] [answered]
### Is your feature request related to a problem? Please describe.

My character have several playsnd controllers, that playing voice in one state. If one contoller activated, then the second should not activate. On current moment I solve the problem using a variable:
```
[State 1300, Playsnd 1]
type = playsnd
trigger1 =  var(1)!=1300
trigger1 = <desired condition>
trigger1 = var(1):=1300
value = 540,random%4
channel = 0

[State 1300, Playsnd 2]
type = playsnd
trigger1 =  var(1)!=1300
trigger1 = <desired condition>
trigger1 = var(1):=1300
value = 340,random%4
channel = 0
```
When the value of a variable (var(1)) does not match the state number, the variable is reset. In principle, this method works perfectly. But a special trigger would be more convenient for this. Something like:

`Issoundplaying(340,1)`

Where 340 is group number of the sound. 1 is the index of playing sound. Trigger `Issoundplaying(340,-1)` will return 1 if sound with any index from group 340 is playing.

And just ideas on what triggers can be added for working with sound:

- Trigger showing which audio channel is active (sound is playing) and which is not
- Trigger showing what sound is playing on what channel
- Trigger showing the number of ticks remaining to play the current sound (something like animtime, just for sounds)

Also, on current moment playsnd controller may have only 2 priorities: low or no priority. I suggest introducing a new parameter for playsnd. Something like **sndpriority**. Similiar to sprpriority, only for sounds. Sound with higher priority can interrupt sound with lower priority. And can't be interrupted by sound with lower priority.

### Search terms

sound, trigger

**Comment by Eiton** [comment]
Sound priority is available since 0.99
https://github.com/ikemen-engine/Ikemen-GO/wiki/State-controllers-(changed)#priority

**Comment by K4thos** [comment, marked as answer]
implemented in #2273

**Comment by Kiwi1444** [comment]
How about the ability to add an echo parameter to any sound, not only KO sound?
